### PR TITLE
[FIX] Build the trimmed maps support sign-agnostically

### LINKED ISSUE #6443: [BUG] _trim_maps builds the support with > 0, so a kept negative-weighted map covers no voxel
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating system

- [x] Linux

### Operating system version

Linux (Ubuntu 22.04, WSL2)

### Python version

- [x] 3.12

### nilearn version

dev (`main` @ be14e46)

### Expected behavior

A map that survives the trim should contribute its voxels to the returned support mask, regardless of the sign of its weights — and `NiftiMapsMasker` should extract a signal for it.

Which maps are kept is decided sign-agnostically, with `abs()` (`nilearn/regions/signal_extraction.py:757`):

```python
sums = abs(maps[as_ndarray(mask, dtype=bool), :]).sum(axis=0)
...
    if not keep_empty and sums[n] == 0:
        continue
```

So a negative-weighted map is kept, and the support should agree.

### Current behavior & error messages

The support is built sign-*sensitively*, three lines below the `abs()` (`signal_extraction.py:773`):

```python
maps_mask[trimmed_maps[..., p] > 0] = 1
```

A negative-weighted map therefore passes the `abs()` filter — it is reported in `indices`, so it is declared present — but contributes no voxel to `maps_mask`. `lstsq` then gets an all-zero column for it, and the extracted signal is silently `0`.

`_trim_maps` directly:

```
weight=+3.0   kept_indices=[0, 1]   support_voxels=2
weight=-3.0   kept_indices=[0, 1]   support_voxels=1     <- region 0 kept but covers nothing
```

End to end through the public `NiftiMapsMasker`:

```
weight=+3.0   ->  signals=[-0.333  0.28 ]
weight=-3.0   ->  signals=[ 0.     0.28 ]     <- region 0's signal is silently zero
```

Only the sign of one map's weight differs between the two rows. Region 1 (always positive) is unaffected in both, which shows the fixture is sound.

This matters for `NiftiMapsMasker` in particular, since its documented targets — ICA components and statistical maps — are *signed*. The docstring for the returned mask describes it as marking where a map is "non-zero", which `> 0` does not implement.

### Steps and code to reproduce bug

```python
import numpy as np
from nilearn.regions.signal_extraction import _trim_maps

mask = np.ones((3, 3, 3), dtype=bool)

for weight in (3.0, -3.0):
    maps = np.zeros((3, 3, 3, 2))
    maps[0, 0, 0, 0] = weight   # region 0: one voxel, this weight
    maps[1, 1, 1, 1] = 5.0      # region 1: always positive (control)
    _, maps_mask, indices = _trim_maps(maps, mask)
    print(f"weight={weight:+.1f}  kept_indices={list(indices)}  support_voxels={int(maps_mask.sum())}")
```

Output:

```
weight=+3.0  kept_indices=[0, 1]  support_voxels=2
weight=-3.0  kept_indices=[0, 1]  support_voxels=1
```

The `+3.0` row is the control: same geometry, same code path, only the sign differs.

---

Disclosure, per AGENTS.md: this issue was found and written by an AI coding agent (Claude Code) running on this account. The repro above was executed against unmodified `main`; the output shown is real and re-runnable.

I have a one-line fix and a test ready (`maps_mask[trimmed_maps[..., p] != 0] = 1`, matching the `abs()` used to decide which maps are kept) and will open a PR referencing this issue if that's welcome.

### PR DESCRIPTION
**Pull request by chuenchen309**
- Closes #6443

Use of AI tools:
- [x] LLM tools were used to generate at least part of the content of this pull-request.
- [ ] No LLM tools were used to generate any part of the content of this pull-request.

Changes proposed in this pull request:

- Build the `_trim_maps` support with `!= 0` instead of `> 0`, so a map kept by the sign-agnostic `abs()` filter actually covers its voxels.
- Add a regression test parametrized over a positive and a negative weight.

---

### The asymmetry

`_trim_maps` decides which maps survive sign-agnostically:

```python
sums = abs(maps[as_ndarray(mask, dtype=bool), :]).sum(axis=0)
...
    if not keep_empty and sums[n] == 0:
        continue
```

but builds their support three lines below with a sign-sensitive test:

```python
maps_mask[trimmed_maps[..., p] > 0] = 1
```

So a negative-weighted map passes the `abs()` filter and is reported in `indices` — declared present — while contributing no voxel to `maps_mask`. `lstsq` then gets an all-zero column for it and the extracted signal is silently `0`.

```
weight=+3.0   kept_indices=[0, 1]   support_voxels=2
weight=-3.0   kept_indices=[0, 1]   support_voxels=1
```

End to end through `NiftiMapsMasker`:

```
weight=+3.0   ->  signals=[-0.333  0.28 ]
weight=-3.0   ->  signals=[ 0.     0.28 ]
```

Only the sign of one map's weight differs between the rows. The always-positive second region is unaffected in both, so the fixture is sound.

This matters for the documented use of `NiftiMapsMasker`: ICA components and statistical maps are signed. The docstring describes the returned mask as marking where a map is "non-zero", which `> 0` does not implement.

### Why it held until now

Every existing `_trim_maps` test uses positive weights only (`1`, `1.1`, `0.5`, `1.4`, ...), so the two halves never disagreed.

### Verification

- New test fails on `main` for the negative parameter and passes with the fix; the positive parameter passes either way.
- `nilearn/regions/tests/test_signal_extraction.py`: 34 passed.
- `nilearn/maskers/tests/test_nifti_maps_masker.py`: 4 failed, 133 passed — those four `check_masker_generate_report` failures are pre-existing and fail identically on unmodified `main` with the same command.
- `ruff check` and `ruff format --check` clean.

---

Disclosure, per AGENTS.md: this contribution was found, written, verified, and described by an AI coding agent (Claude Code) running on this account. Every figure above came from actually running the code rather than reading it — but the agent ran it, not a person. The new test carries `@pytest.mark.ai_generated` as AGENTS.md asks.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
converting to draft until we have figured out if negative values in a map image is a realistic scenario

**Comment by chuenchen309** [comment]
Yes — signed maps are the normal input here, so this is a real scenario:

- **The function already assumes it.** `_trim_maps` selects the surviving maps with `abs(maps[...]).sum(axis=0)` — a sign-agnostic filter. If negative values weren't expected there, that `abs()` would be unnecessary. The `> 0` support line three rows below is simply inconsistent with the `abs()` the function already commits to.
- **The documented inputs are signed.** `NiftiMapsMasker` is the standard masker for ICA / dictionary-learning components (`CanICA`, `DictLearning`) and for statistical maps — all of which carry negative values by construction (a component's sign is arbitrary; deactivations are negative). The docstring describes `maps_mask` as marking where a map is "non-zero", which `> 0` does not implement.

So a user passing a component with negative loadings gets it reported in `indices` but silently extracted as `0`. Happy to add a docstring line or extend the test if that helps confirm the scenario.

**Review by bthirion — COMMENTED** [review]
LGTM. Can you fix the doc build failure ?

File: nilearn/regions/tests/test_signal_extraction.py [hunk 1]
```diff
@@ -799,6 +799,27 @@ def test_trim_maps_half_regions(shape_3d_default):
     assert_equal(np.asarray(list(range(4))), maps_i_indices)
 
 
+@pytest.mark.ai_generated
```
Comment (Remi-Gau) [inline nilearn/regions/tests/test_signal_extraction.py]:
```suggestion
```

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
> Yes — signed maps are the normal input here, so this is a real scenario:
> 
>     * **The function already assumes it.** `_trim_maps` selects the surviving maps with `abs(maps[...]).sum(axis=0)` — a sign-agnostic filter. If negative values weren't expected there, that `abs()` would be unnecessary. The `> 0` support line three rows below is simply inconsistent with the `abs()` the function already commits to.
> 
>     * **The documented inputs are signed.** `NiftiMapsMasker` is the standard masker for ICA / dictionary-learning components (`CanICA`, `DictLearning`) and for statistical maps — all of which carry negative values by construction (a component's sign is arbitrary; deactivations are negative). The docstring describes `maps_mask` as marking where a map is "non-zero", which `> 0` does not implement.
> 
> 
> So a user passing a component with negative loadings gets it reported in `indices` but silently extracted as `0`. Happy to add a docstring line or extend the test if that helps confirm the scenario.

Forgot that the maps maskers are used as embedded maskers for things like ICA, so yeah negative maps make sense
